src/geometry/adjoints.jl

```diff
@@ -49,3 +49,50 @@ function ChainRulesCore.rrule(::Type{SVector{N, T}}, x...) where {N, T}
     end
     return y, svector_const_pb
 end
+
+function ChainRulesCore.rrule(::typeof(ImageTransformations._getindex), A::AbstractExtrapolation, x)
+    y = A(Tuple(x)...)
+    function _getindex_pb(Δy)
+        # Δy :: NamedTuple{(:r, :g, :b)} or something similar
+        Δy = eltype(A)(Δy...)
+        gr = Interpolations.gradient(A, Tuple(x)...)
+        n = nfields(Δy) > 0 ? nfields(Δy) : 1
+        return NoTangent(), NoTangent(), Δy .⋅ gr .⋅ n
+    end
+    return y, _getindex_pb
+end
+
+function ChainRulesCore.rrule(h::DiffImages.Homography, x::SVector{2,K}) where {K}
+    # elements of x => (i₀,j₀)
+    y = h(x) # Gives us (i j)
+    function Homography_pb(Δy)
+        # Δy => (∂g/∂i ∂g/∂j) :: SVector{2, Float64}
+        v₁ = h.H * SVector((x..., 1.0)) # Gives us (i₁ j₁ z)
+        ∂ij_∂W = begin
+            r₁ = [x[1] x[2] 1]
+            r₂ = [0.0 0.0 0.0]
+            [1 / v₁[3] * vcat(r₁, r₂, -y[1] * r₁), 1 / v₁[3] * vcat(r₂, r₁, -y[2] * r₁)]
+        end
+        return Tangent{DiffImages.Homography}(H = Δy' * ∂ij_∂W), (h.H[1:2, 1:2]*Δy) ./ v₁[3]
+    end
+    return y, Homography_pb
+end
+
+function ChainRulesCore.rrule(::typeof(ImageTransformations.warp!), out, img::AbstractExtrapolation, tform)
+    out = ImageTransformations.warp!(out, img, tform)
+    function warp!_pb(Δy)
+        ∇out = NoTangent()
+        ∇img = @not_implemented("To be implemented.")
+        ∇tform = ZeroTangent() # Change this to Tangent{primal}
+        Δy = collect(Δy)
+        for p in CartesianIndices(out)
+            _, ∇τ = rrule(ImageTransformations._getindex, img, p.I)
+            _, _, ∇τ = ∇τ(Δy[p])
+            _, ∇ϕ = rrule(tform, SVector(p.I))
```

Comment:
Use `rrule_via_ad` here.

`rrule_via_ad` for these methods needs to forward to `Zygote.pullback`